pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -651,3 +711,137 @@ func getMinTime(t1, t2 time.Duration) time.Duration {
+	}
+	listener, err := net.Listen("unix", exp.l7ListenerSocketPath)
+	if err != nil {
+		klog.ErrorS(err, "Error listening on Suricata socket")
+		return
+	}
+	defer listener.Close()
+
+	klog.InfoS("L7 Listener Server started. Listening for connections...")
+
+	for {
+		conn, err := listener.Accept()
+		if err != nil {
+			klog.ErrorS(err, "Error accepting Suricata connection")
+			continue
+		}
+		go func() {
+			exp.handleClientConnection(conn)
+		}()
+	}
+}
+
+func (exp *FlowExporter) handleClientConnection(conn net.Conn) {
+	defer conn.Close()
+	reader := bufio.NewReader(conn)
+	for {
+		buffer, err := reader.ReadBytes('\n')
+		if err != nil && err != io.EOF {
+			klog.ErrorS(err, "Error reading data", "buffer", buffer)
+			return
+		}
+		exp.processLog(buffer)
+	}
+}
+
+func (exp *FlowExporter) processLog(data []byte) {
+	exp.l7mut.Lock()
+	defer exp.l7mut.Unlock()
+	// Check if the event type is "http"
+	if eventType := extractEventType(data); eventType != "http" {
+		return
+	}
+	var event JsonToEvent
+	err := json.Unmarshal(data, &event)
+	if err != nil {
+		klog.ErrorS(err, "Error parsing JSON:", "data", data)
+		return
+	}
+	exp.WriteConnection(event)
+}
+
+func extractEventType(data []byte) string {
+	// Define a temporary struct for extracting the event type
+	type Temp struct {
+		EventType string `json:"event_type"`
+	}
+
+	// Parse the JSON string and extract the event type
+	var temp Temp
+	err := json.Unmarshal(data, &temp)
+	if err != nil && !strings.Contains(err.Error(), "unexpected end of JSON input") {
+		klog.ErrorS(err, "Error parsing JSON for eventtype:", "data", data)
+		return ""
+	}
+
+	return temp.EventType
+}
+
+func (exp *FlowExporter) WriteConnection(event JsonToEvent) {
+	protocol, err := flowexporter.LookupProtocolMap(event.Proto)
+	if err != nil {
+		klog.ErrorS(err, "InValid Protocol type")
+		return
+	}
+	// Get 5-tuple information
+	tuple := flowexporter.Tuple{
+		SourceAddress:      event.SrcIP,
+		DestinationAddress: event.DestIP,
+		Protocol:           protocol,
+		SourcePort:         uint16(event.SrcPort),
+		DestinationPort:    uint16(event.DestPort),
+	}
+	conn := flowexporter.Connection{}
+	conn.FlowKey = tuple
+	connKey := flowexporter.NewConnectionKey(&conn)
+	var tempArr []string
+	tempArr = append(tempArr,
+		event.HTTP.Hostname,
+		event.HTTP.URL,
+		event.HTTP.UserAgent,
+		event.HTTP.ContentType,
+		event.HTTP.Method,
+		event.HTTP.Protocol,
+		strconv.Itoa(event.HTTP.Status),
+		strconv.Itoa(event.HTTP.ContentLength))
+	if _, found := exp.l7events[connKey]; found {
+		exp.l7events[connKey].http = append(exp.l7events[connKey].http, tempArr)
+	} else {
+		var temp httpEvent
+		temp.http = append(temp.http, tempArr)
+		exp.l7events[connKey] = &temp
+	}
+}
+
+func (exp *FlowExporter) getL7EventData(connkey flowexporter.ConnectionKey, field string) string {
+	exp.l7mut.Lock()
+	defer func() {
+		exp.l7mut.Unlock()
+	}()
+	_, exists := exp.l7events[connkey]
```

Comment:
nit: typically we use `ok` as the variable name (instead of `exists`), by convention